app/components/whappubuddy/BuddyUserView.js

```diff
@@ -78,15 +75,13 @@ class BuddyUserView extends Component {
     } else {
       this.props.fetchBuddyProfile(userId);
     }
-    
+
   }
 
   componentWillReceiveProps({ tab, userId }) {
     // Fetch images and data on Buddy tab if this is the user's own profile
     if (tab !== this.props.tab && tab === 'BUDDY') {
-      this.props.fetchUserImages(userId);
-      this.props.fetchUserProfile(userId);
-      this.props.fetchBuddyProfile(userId);
+      this.props.fetchBuddyProfile(this.props.buddies.get(this.state.buddyIndex).id);
```

Comment:
If the user has no WhappuBuddy profile, i.e. has not registered for it yet, this line will cause an error, saying that that it cannot find the property _id_ of undefined. No profile is then fetched to be displayed.

As a fix I would suggest trying to fire this call only if the usesWhappuBuddy prop from the [feature/buddyIntro branch](/Zackyist/wappuapp-client/tree/feature/buddyIntro) is true. It validates that the user has actually registered on WhappuBuddy.

But there also needs to be a way for unregistered people to browse potential buddies. They should get an unfiltered list of all the registered Buddy users which they can browse with the Skip button. The opinion buttons will then be visible but will not trigger a like or dislike (dealing with the opinion buttons is a part of the feature/buddyIntro branch so no need to worry about it here).